app/models/skin.rb

```diff
@@ -498,6 +516,7 @@ def self.create_default
     end
     skin.official = true
     skin.save!
+    Rails.cache.increment('skins_generation/'+(skin.type.nil? ? ("site_skin") : skin.id.to_s) )
     skin
   end
   
```

Comment:
So we have site_skins which all share the same cache key, when any of those change the cache key will change.

And work_skins which do not have parents, so I think this can be.

the style block cache doesn't expire when nothing has been changed
the style block cache expires when the skin is edited
the work skin style blocks don't expire when a site skin is edited
the  style block cache doesn't expire when a site skin is edited

However those are quiet hard to write as tests and given the one line of code pretty self evident.

What do you think ?